daemon/api_notices.go

```diff
@@ -47,6 +53,14 @@ var (
 	}
 )
 
+const (
+	maxNoticeKeyLength = 256
```

Comment:
Should this be a more common place for notices? I'm not sure I feel like this would be the correct place to have it. Also do we have some more notice-key validation logic somewhere it could fit?

Reply:
I moved the validation to `state/notices.go` and shared it with the daemon, I like this better, Thanks!